docs/docs/dagster-basics-tutorial/6-custom-components.md

```diff
@@ -0,0 +1,109 @@
+---
+title: Components
+description: Defining our own components
+sidebar_position: 90
+---
+
+So far we have built out everything via `Definitions`. But this is not the only way to include `Definitions` in a Dagster project.
+
+If we think about the code for our three assets (`customers`, `orders`, and `payments`), it is all very similar. Each asset performs the same action, turning an S3 file into a DuckDB table while differing only in the URL path and table name.
+
+These assets are a great candidate for a [component](/guides/build/components/creating-new-components). Components generate `Definitions` through a configuration layer. There are built-in components that let you integrate with common workflows (such as turning Python scripts into assets) or with popular tools like [dbt](https://www.getdbt.com/) or [Fivetran](https://www.fivetran.com/). There are also custom components that let you define your own specific use cases.
+
+Here we will use a custom component to streamline the development of our assets and replace their `Definitions` in our project with a `Component` which can generate them instead.
+
+![2048 resolution](/images/tutorial/dagster-tutorial/overviews/components.png)
+
+## 1. Scaffold a custom component
+
+First, scaffold a custom component using `dg`:
+
+<CliInvocationExample path="docs_snippets/docs_snippets/guides/tutorials/dagster-tutorial/commands/dg-create-custom-component.txt" />
+
+This adds a new directory, `components`, within `src/dagster_tutorial`:
+
+<CliInvocationExample path="docs_snippets/docs_snippets/guides/tutorials/dagster-tutorial/tree/step-6a.txt" />
+
+This directory contains the files needed to define the custom component.
+
+## 2. Define the custom component
+
+When designing a component, keep its interface in mind. In this case, the assets that our component will create share the following attributes:
+
+- A DuckDB database shared across all assets.
+- A list of ETL assets, each with a URL path and a table name.
+
+The first step is to create a `dg.Model` for our ETL assets. These are similar to [Pydantic](https://docs.pydantic.dev/) models. This model will contain the two attributes that define an asset:
```

Comment:
```suggestion
The first step is to create a `dg.Model` for the ETL assets. These are similar to [Pydantic](https://docs.pydantic.dev/) models. This model will contain the two attributes that define an asset:
```

Reply:
Minor nit, I think it's more accurate to say `dg.Model` turns any class that inherits from it into a Pydantic model. (`Model` is also used to implement a YAML interface for the component.)